### DISCUSSION TITLE
Hardware question

### DISCUSSION THREAD

**Discussion by inferno1231** [awawa-dev/HyperHDR#267] [General]
Hello,

I am currently running SK6812RGBNW setup with 332 LED's + ESP32 (CP2102) running pure WLED + HyperHDR v17 on Pi. 

I want to try a USB solution with HyperHDR v18 beta, so I ordered ESP32 with CH340g onboard (CH9102x are impossible to find in Europe it seems). There is one thing I can't understand - right now my ESP is connected via micro-usb cable to my TV's USB, I also run one jumper wire from GPIO2 to my strip for data and one jumper wire from GND to same strip's ground for common ground. I am not using any level shifters and it seems to work OK. 

So, when I will flash ESP32 with HyperSerialESP32, I will run a micro-USB cable from ESP to Pi, I will leave GND and Data wires in place.. where and why do I need a level shifter? Should I run extra wire from 3.3 V pin of ESP to strip? If yes, why don't I need it now? 

Sorry for stupid question, I just started to get into this like 3 days ago, and I never had any experience before. 

Thank you in advance :)

**Comment by awawa-dev** [comment]
Hi
If the current setup works OK, without blinking occasionally, it should probably work if you just switch ESP from TV to Rpi, even if you switch from wireless to USB communication. Old wiring from ESP to LED strip will not be changed / no modification of the cable length, because it may be important? The LED strip does not need the 3.3V power supply of your ESP, only the common ground and data lines that are already connected.

I think this is a 99% chance of success without any changes to the hardware configuration. Why not 100%? Since your data line is already running around 3.3V, which is too low for an LED strip, but it works. If, after switching ESP from the TV, Rpi delivers a slightly lower + 5V via USB (unstable Rpi power supply or longer USB cable), and the ESP for some reason does not compensate it properly, the data line may no longer be tolerated. In this case, the level lever may become necessary.

**Comment by awawa-dev** [comment]
BTW CP2102 is capable of 921600 speed. You can test your setup and USB connection already using latest HyperSerialWLED and changing speed in WLED (it's configurable now) and HyperHDR to 921600 (AWA protocol must be enabled of course). And after acquiring CH340g you can switch to 2Mb speed.

**Reply by inferno1231** [reply]
That's a nice idea. By the way, how huge are visual improvements when switching from default 115200 to 921000 and from 921000 to 2000000?

**Reply by awawa-dev** [reply]
The higher number of LED in the strip the more important it is and you have quite a big setup. 2000000 allows the SK6812 data line to be almost fully saturated and to reach the maximum refresh rate. Almost fully saturated: because of the communication overhead there is still some room for improvement with HyperSPI or 4Mb when CH9102x is used. 921000 means almost 2 times worse result. 115200  is good for slide show or static effects 😉   
The higher the refresh rate, the better the experience. Even if the video source/grabber provides 30FSP video content, HyperHDR smoothing can generate a much higher refresh rate which means smoother transitions between scenes.

**Comment by inferno1231** [comment]
Data wire from ESP to strip will be the same, well, in fact, I can make it a little bit shorter I think. USB cable will be much longer unfortunately. Right now it is something like 15-20 cm., and if I will connect Pi to ESP it will be like 2 m. cable. Seems like I should do some testing.

**Comment by inferno1231** [comment]
Thank you very much for the answers, @awawa-dev . If I may, one more question about grabber and settings in HyperHDR. I am using cheap USB 3.0 grabber with, according to lsusb command, some kind of Macrosilicon chip inside. Is there any way to find out which one it specifically is without disassembling the unit? Or should I just put data into HyperHDR according to tests from blog?

**Comment by awawa-dev** [comment]
You don't need to disassembling it. 
Type ```usb-devices```
If its MS2109 then you should receive VendorId=534d, ProdId=2109  
MS2109 doesn't really support USB3.0, At least I haven't met such version.

![link](https://i.postimg.cc/sf7PT2s6/ms-1.jpg)

If you need faster grabber then probably Ezcap 321 could be a good choice.

**Reply by inferno1231** [reply]
> 
How do I check if it is really USB 3.0? I can capture YUV.

![htgCqLG](https://user-images.githubusercontent.com/25138913/167171993-dd6cd3af-e14c-45a8-90e2-18401b284697.png)

**Comment by awawa-dev** [comment]
Speed(Spd) is 480: that's USB 2.0 characteristic. The manufacturer would needed to make a serious modifications if it's possible at all because MS2109 has built-in USB2.0 controller. http://en.macrosilicon.com/info.asp?base_id=2&third_id=50  
The safe assumption is that these MS2109 clones sold as USB3.0 are a scam and it's best to choose models based on the first generation design (solid metal can, like Rullz on my blog). Later it was only worse: fake USB3.0,  problems with overheating due to bad custom board designs or because of the removed heat sinks (to reduce cost).

**Comment by inferno1231** [comment]
Ok, I got some time to play around with new CH340C ESP32's and I miserably failed. @awawa-dev I need Your help again :(

I upgraded to HyperHDR v18 beta2.
So, I flashed new board with HyperSerialESP32, connected it directly to Pi using USB cable. Selected output path, AWA, 2000000 baud rate and nothing happened. 
Ok, I took one of my old CP2102, flashed it with HyperSerialWLED, set baudrate to 921600, and nothing happened again, LED's just don't light up at all. Dead.

I took another CP2102, flashed it with vanilla WLED, 115200 baudrate, same in HyperHDR, adalight, AWA disabled - does not work. In this case it lights up, when I restart HyperHDR/ESP32, an effect which is preset in HyperHDR works on startup, after that I get like a rainbow around whole strip and then it turns off fully. I can see live feed from grabber, I can see leds on live feed, that they change etc - but those don't work irl.

If I start using WLED wireless connection, everything works fine, doesn't matter if I power ESP from Pi or from TV usb port.
All parts common grounded.
Pi is powered using separate original power supply.
dmesg shows no errors.
HyperHDR logs looking clear too.
I tried 2 different usb cables, different ports on Pi.
I am out of ideas.

Also, if I will have to stick to wireless connection, what is max possible baudrate when using it?

Many thanks in advance.

**Comment by awawa-dev** [comment]
OK, since your LED strip is working when using wifi it's not that bad. Forget about HyperSerialESP32 for now and flash CH340C with HyperSerialWLED again for testing. Make sure you have latest HyperSerialWLED v7. Previous version had a problem with saving serial port speed due to some changes in WLED (they introduced artificial limit on start up). Set 921600 or 512000 speed both in HyperSerialWLED  and in HyperHDR . It must work using AWA protocol (RGBW calibration in the adalight driver properties in HyperHDR must be DISABLED for HyperSerialWLED ).

> I took one of my old CP2102, flashed it with HyperSerialWLED, set baudrate to 921600, and nothing happened again, LED's just don't light up at all. Dead.  

Did you test WLED effects using WLED interface? Did WLED page was displaying the information about receiving live USB data from HyperHDR?  

>I took another CP2102, flashed it with vanilla WLED, 115200 baudrate, same in HyperHDR, adalight, AWA disabled - does not work. In this case it lights up, when I restart HyperHDR/ESP32, an effect which is preset in HyperHDR works on startup, after that I get like a rainbow around whole strip and then it turns off fully. I can see live feed from grabber, I can see leds on live feed, that they change etc - but those don't work irl.

Quite strange, did you try to start an effect from the remote tab? Because it's like the grabber source was not active or the signal detection was triggered. Search for active priority in the logs. Grabber has 240 on default. Was WLED page  displaying the information about receving data from HyperHDR all the time?

And now the tricky question: was the ESP always connected to Rpi even when you used wireless communication?

**Comment by awawa-dev** [comment]
> I took another CP2102, flashed it with vanilla WLED, 115200 baudrate, same in HyperHDR, adalight, AWA disabled - does not work. In this case it lights up, when I restart HyperHDR/ESP32, an effect which is preset in HyperHDR works on startup, after that I get like a rainbow around whole strip and then it turns off fully. I can see live feed from grabber, I can see leds on live feed, that they change etc - but those don't work irl.

Maybe it's related? [link https://github.com/Aircoookie/WLED/issues/2618](https://github.com/Aircoookie/WLED/issues/2618)

**Reply by inferno1231** [reply]
> Maybe it's related? [link https://github.com/Aircoookie/WLED/issues/2618](https://github.com/Aircoookie/WLED/issues/2618)

Well, it could be. There were 1 or 2 times when I switched from wled to adalight and my ESP stopped responding until restart.

**Reply by awawa-dev** [reply]
So probably it's better to stick with HyperSerialWLED: I've rewritten all the serial communication in my fork. Unless it's related to some internal LED routines maybe it's not affected.

**Reply by inferno1231** [reply]
> So probably it's better to stick with HyperSerialWLED: I've rewritten all the serial communication in my fork. Unless it's related to some internal LED routines maybe it's not affected.

Ok, I will move my setup to CH340C with HyperSerialWLED, if adalight won't work, I will simply switch to wireless connection. I saw possibility to choose baudrate up to 2000000. Is it worth to use increased baudrate when connecting wireless?

**Reply by awawa-dev** [reply]
it must work at 2Mb or lower speed using serial adalight AWA protocol: I don't see other option even if it's counterfeited chip. When wireless connection is used, the baudrate for serial communication is just ignored.

**Reply by awawa-dev** [reply]
when you change baudrate for USB serial communication you must change it in both: HyperHDR adalight driver and in HyperSerialWLED (using WLED page)

**Comment by inferno1231** [comment]
>OK, since your LED strip is working when using wifi it's not that bad. Forget about HyperSerialESP32 for now and flash CH340C with HyperSerialWLED again for testing. Make sure you have latest HyperSerialWLED v7. Previous version had a problem with saving serial port speed due to some changes in WLED (they introduced artificial limit on start up). Set 921600 or 512000 speed both in HyperSerialWLED and in HyperHDR . It must work using AWA protocol (RGBW calibration in the adalight driver properties in HyperHDR must be DISABLED for HyperSerialWLED ).

Ok, I will try. If I remember correctly, I tried with AWA protocol disabled. 

>Did you test WLED effects using WLED interface? Did WLED page was displaying the information about receiving live USB data from HyperHDR?

No, will do.

>Search for active priority in the logs. Grabber has 240 on default. Was WLED page displaying the information about receving data from HyperHDR all the time?

Yes, when using vanilla WLED (usb connection) I saw that message, but LED's were not lighting up after startup effect.

>And now the tricky question: was the ESP always connected to Rpi even when you used wireless communication?

Yes, all tests were done powering ESP from RPi's USB port.

**Reply by awawa-dev** [reply]
OK, try again. For HyperSerialWLED  AWA option must be enable. In v18beta2 there is also new option for calibration white channel for RGBW LED strip but it must be disabled for HyperSerialWLED  (WLED does its own calibration anyway).

**Comment by inferno1231** [comment]
Looks like I am out of luck and my ESP32 - C3 is not yet fully supported by WLED.

**Comment by awawa-dev** [comment]
@inferno1231 
Take a look here: https://github.com/awawa-dev/HyperSerialWLED/actions/runs/2296235297
It's WLED github action that provides version for ESP32 - C3. But for some reason they didn't decide to include it in the official release and I didn't test it either.

**Comment by inferno1231** [comment]
@awawa-dev Ok, thank you! Right now I am playing with CP2102 ESP32. I flashed HyperSerialWLED v7, connected to my setup. WLED Effects work when using adalight / AWA protocol. But I can't connect HyperHDR and ESP, there is no message that WLED is receiving data from HyperHDR. Also in "Overview" - Components - LED Device is off. But I can get a live feed no problem.
<img width="798" alt="Screenshot 2022-05-17 at 00 58 55" src="https://user-images.githubusercontent.com/25138913/168689220-41b6fce1-4d07-4edf-a657-4555918e56a6.png">
Looks like I am missing something. 
<img width="743" alt="Screenshot 2022-05-17 at 00 59 57" src="https://user-images.githubusercontent.com/25138913/168689332-084bcbe1-216f-419c-a08a-609acbc83c7d.png">

**Reply by awawa-dev** [reply]
ttyUSB0 is valid path? What's happen when you click 'output path' button? If the path is valid then try to set 512000 for test.

> "Overview" - Components - LED Device is off  

That's bad. Check logs for red errors because I suspect wrong serial port path. If there is no error then the only explanation is no video source: try to play some effects from the HyperHDR remote tab.

**Reply by inferno1231** [reply]
> ttyUSB0 is valid path? What's happen when you click 'output path' button? If the path is valid then try to set 512000 for test.
> 
> > "Overview" - Components - LED Device is off
> 
> That's bad. Check logs for red errors because I suspect wrong serial port path. If there is no error then the only explanation is no video source: try to play some effects from the HyperHDR remote tab.

I restarted instance in HyperHDR, now LED Device is ON. 
Effects do not work. 
Regarding path.. I guess..
<img width="1039" alt="Screenshot 2022-05-17 at 01 13 51" src="https://user-images.githubusercontent.com/25138913/168691054-b4eaca8f-4547-4d85-82e1-48d1d2a7135b.png">

If I choose CP2102 it goes back to "Output path" after refresh.
<img width="686" alt="Screenshot 2022-05-17 at 01 15 40" src="https://user-images.githubusercontent.com/25138913/168691228-452b5abf-be73-4454-b6ce-bc0137990011.png">

 Logs: https://pastebin.com/ZP0zmZY2

Yeah, and it seems ESP drops every 10 minutes or so. Have to restart it constantly. 

Update: When I disabled instance, suddenly WLED picked up USB signal from HyperHDR. And after that WLED/ESP crashed again. I am not sure, but maybe it is related to my USB cables?

**Reply by awawa-dev** [reply]
> If I choose CP2102 it goes back to "Output path" after refresh.  

that's expected: it's only for choosing path on the left.

Logs looks OK: there are not errors, the grabber is active and providing the content. Also LED device is enabled this time. WLED should not crash under any circumstances: probably that previously mentioned WLED bug causes it or it's some hardware/power supply instability.  That version with CP2102 is ESP32 or ESP8266? You can try older special WLED release that has hardcoded 921600 speed from here: https://github.com/awawa-dev/HyperSerialWLED/releases It's stable (as for WLED)

**Reply by awawa-dev** [reply]
sorry that link https://github.com/awawa-dev/HyperSerialWLED/releases/tag/v3.0.0.0

**Comment by inferno1231** [comment]
It works with 500000 baudrate! Looks like my ESP can't handle 921600. I think I will wait till my new CH9102X will arrive from Aliexpress and try with them. Hopefully those will be able to handle high baudrate. I am sorry for wasting your time and grateful for the help :)

Or it could be USB cable rather than ESP?

**Reply by awawa-dev** [reply]
CP2102 definitively should support 921600. It's proven chip. Rather some hardware problems with the board or low quality board...921600~1Mb is very low as for USB2.0 transfer so the cable probably doesn't cause it. Are you sure you have CP2102 ?

**Reply by inferno1231** [reply]
> CP2102 definitively should support 921600. It's proven chip. Rather some hardware problems with the board or low quality board...921600~1Mb is very low as for USB2.0 transfer so the cable probably doesn't cause it. Are you sure you have CP2102 ?

Well.. I got them here - https://www.az-delivery.de/en/products/esp32-developmentboard?variant=36542176914

I can see CP2102 on chip visually.

Question about grabber - should I use YUYV 30 FPS versus MJPEG 60 FPS?

**Reply by awawa-dev** [reply]
> I can see CP2102 on chip visually.  

One more possibility: latest WLED crashes because of high refresh rate and it's related to USB serial port speed/data transfer from HyperHDR

> Question about grabber - should I use YUYV 30 FPS versus MJPEG 60 FPS?

Give the try both and read statistics in the log. Something like that ( decoding = 2ms and if FPS is reaching 30FPS or 60FPS depending on your settings)
```2022-05-16T22:18:00.034Z [PERFORMANCE] [USB capturing: FPS = 30.02, decoding = 2ms, frames = 1802, invalid = 0], [INSTANCE0: FPS = 30.04, processed = 1802], [LED0: FPS = 25.00, send = 1500, processed = 1500]```
Also watch CPU load in the overview tab or using htop (turn on HDR tone mapping as it is using some resources). YUV processing should be faster and lighter but MJPEG's 60FPS also is a factor. v18beta2 is well optimized for mjpeg.

**Reply by inferno1231** [reply]
> One more possibility: latest WLED crashes because of high refresh rate and it's related to USB serial port speed/data transfer from HyperHDR

Yeah, something is definitely wrong with board.. I am testing automatic detection now, and it turns off perfectly, but when trying to turn on, WLED does not come online.

I hope new board will resolve this issues, or maybe I will try HyperSerialESP32 tomorrow with lower baudrate.

**Comment by inferno1231** [comment]
Ok, I got my CH9102X today and spent some time testing everything.
So, maybe my experience will help someone.

Firstly I installed CH343 driver to RPi.

As @awawa-dev mentioned before, there is indeed a problem with latest WLED 0.13.1 version (including HyperSerialWLED v6.0.0.0 and v7.0.0.0), after switching to adalight ESP32 turns dead. I flashed my ESP32 with HyperSerialWLED v5.0.0.0beta and everything started working with 2000000 baudrate.

After that I came back to my initial plan using HyperSerialESP32, and flashed latest pre-release version (v6.0.0.0). LEDs did not light up.
I checked using `screen` command and got this: https://i.imgur.com/F6pjEkm.png

After that I flashed HyperSerialESP32 v5.0.0.0 and everything lightened up again using 2000000 baudrate (I did not activate "White channel calibration (RGBW only)". 

So, is there any way to find out why v6 doesn't work on my setup?

My target is to use 4MB baudrate, so I am not sure, if I should compile v5 with 4000000 or somehow fix v6 :)

**Comment by awawa-dev** [comment]
> after switching to adalight ESP32 turns dead  

what's exactly happen: your LED didn't work with HyperHDR or WLED page wasn't accessible? After flashing HyperSerialWLED **v7.0.0.0** (yes, v7 it's very important) and possible restarting it, did you check what's your serial port speed in WLED page? Set it manually in WLED to 2000000 and give it a try again. I'm a little bit sick of WLED: they had full year of beta releases and that latest stable version has even a lot more seriuos bugs than previous betas. 

> After that I came back to my initial plan using HyperSerialESP32, and flashed latest pre-release version (v6.0.0.0). LEDs did not light up.
I checked using screen command and got this: https://i.imgur.com/F6pjEkm.png  

something was sent but it was a garbage

**Reply by inferno1231** [reply]
> what's exactly happen: your LED didn't work with HyperHDR or WLED page wasn't accessible? After flashing HyperSerialWLED **v7.0.0.0** (yes, v7 it's very important) and possible restarting it, did you check what's your serial port speed in WLED page? Set it manually in WLED to 2000000 and give it a try again. I'm a little bit sick of WLED: they had full year of beta releases and that latest stable version has even a lot more seriuos bugs than previous betas.

Yes, I triple-checked everything, the only way I got it working was setting 115200 in both WLED and HyperHDR, then it lightened up.
Switching to 500000/2000000, WLED page became inaccessible till I physically unplugged and plugged back USB cable. I think it is related to this issue, as you mentioned earlier - https://github.com/Aircoookie/WLED/issues/2618

**Reply by awawa-dev** [reply]
And did you try to activate "White channel calibration (RGBW only)" for HyperSerialESP32 v6?

**Reply by inferno1231** [reply]
> And did you try to activate "White channel calibration (RGBW only)" for HyperSerialESP32 v6?

Yes, I actually have board with v6 hooked to TV right now. 
Here are the settings - https://i.imgur.com/djJzAHZ.png
LEDs are not working.

**Reply by awawa-dev** [reply]
And tell me which exactly firmware for HyperSerialESP32 v6 did you use (preferable link to download it from github to avoid a mistake). I will test it again.

**Reply by awawa-dev** [reply]
> LEDs are not working.  

For HyperSerialESP32  only ```screen``` can confirm the communication problem. But HyperHDR and ```screen``` can't run both at the same time. Run first HyperHDR then close it and run screen.

**Reply by inferno1231** [reply]
> And tell me which exactly firmware for HyperSerialESP32 v6 did you use (preferable link to download it from github to avoid a mistake). I will test it again.

https://github.com/awawa-dev/HyperSerialESP32/releases/download/v6.0.0.0/HyperSerialESP32_SK6812_NEUTRAL_WHITE_FIRSTLED_ENABLED.bin

> For HyperSerialESP32 only `screen` can confirm the communication problem. But HyperHDR and `screen` can't run both at the same time. Run first HyperHDR then close it and run screen.

Ok, just done it one more time - now I get this:
```
-------------------------
HyperSerialESP32 version 6.
Statistics for the last full 1 second cycle.
Frames per second: 0
Good frames: 0
Bad frames:  0
-------------------------
HyperSerialESP32 version 6.
Statistics for the last full 1 second cycle.
Frames per second: 0
Good frames: 0
Bad frames:  0
-------------------------
```

**Reply by awawa-dev** [reply]
I tested it again, works well for 330 leds:
![obraz](https://user-images.githubusercontent.com/69086569/169170693-fdacb953-1b10-418a-9afc-ab64be24dc0b.png)
Unfortunately can't pinpoint the problem in your case: probably configuration of the adalight driver in HyperHDR or something other like busy serial port used by other application like screen (but in that case you should receive a error in HyperHDR). The  HyperSerialESP32 v6 firrmare is fine and it was double checked.

**Reply by awawa-dev** [reply]
PS: these statistics in ```screen``` will be updated only if HyperHDR was using HyperSerialESP32 for a moment before running ```screen```.

**Reply by inferno1231** [reply]
> PS: these statistics in screen will be updated only if HyperHDR was using HyperSerialESP32 for a moment before running `screen`.  

Yep, tried it one more time.

```
-------------------------
HyperSerialESP32 version 6.
Statistics for the last full 1 second cycle.
Frames per second: 29
Good frames: 0
Bad frames:  29
-------------------------
```

**Reply by inferno1231** [reply]
> I tested it again, works well for 330 leds: 
Unfortunately can't pinpoint the problem in your case: probably configuration of the adalight driver in HyperHDR or something other like busy serial port used by other application like screen (but in that case you should receive a error in HyperHDR). The HyperSerialESP32 v6 firrmare is fine and it was double checked.

There are no errors in logs: https://pastebin.com/0cTq7XFM
Adalight settings are as on my screenshot above. 
It works using same settings on v5...

I tried Cold White version with 1 LED Enabled and it is the same.

**Reply by awawa-dev** [reply]
> PS: these statistics in screen will be updated only if HyperHDR was using HyperSerialESP32 for a moment before running `screen`.  
Yep, tried it one more time.

```
-------------------------
HyperSerialESP32 version 6.
Statistics for the last full 1 second cycle.
Frames per second: 29
Good frames: 0
Bad frames:  29
-------------------------
```

You definitely did not check AWA option in HyperHDR: that's typical symptom of missing checksum that was expected and the frame was rejected. But the header was recognized so the communication is probably fine.

**Reply by inferno1231** [reply]
> You definitely did not check AWA option in HyperHDR: that's typical symptom of missing checksum that was expected and the frame was rejected. But the header was recognized so the communication is probably fine.

I did, 100% enabled, and I am still getting bad frames. 
`2022-05-19T00:03:08.343Z [LEDDEVICE_ADALIGHT] (LedDevice.cpp:163) deviceConfig: [{"awa_mode":true,"colorOrder":"rgb","currentLedCount":332,"delayAfterConnect":0,"hardwareLedCount":332,"output":"ttyCH343USB0","rate":2000000,"refreshTime":0,"type":"adalight","white_channel_blue":255,"white_channel_calibration":true,"white_channel_green":255,"white_channel_limit":100,"white_channel_red":255}]`

I also tried disabling AWA and checking with `screen` and (of course) I am getting bad frames.

**Reply by awawa-dev** [reply]
I think I've got something but it's too late tonight to verify it. Will inform you later.

**Comment by awawa-dev** [comment]
Good catch @inferno1231  
I've manged to reproduce the symptom for very large LED setup and high refresh rate (otherwise works fine) using HyperSerialESP32 v6. But then I switched to old HyperSerialESP32 v5 and surprise: it doesn't work for large setups anymore after I built it now but still works for smaller LED strips. And another surprise: old firmware from the github v5 release works fine for the same large LED setup. Tried to increase the inner buffer size but still newer builds were failing.

Finally I found the cause: the Arduino-esp32 board module in Arduino IDE. Previous HyperSerialESP32 v5 firmware was built using Arduino-esp32 version 1.x and HyperSerialESP32 v6 was built using Arduino-esp32 version 2.0.2. Newest 2.0.3 doesn't work properly either. Something must have changed inside that core library since Arduino-esp32 2.0 and it has a grave effect for the serial communication performance at 2Mb speed, causing very large frames become broken. EDIT: Maybe it's related https://github.com/espressif/arduino-esp32/issues/7474 ?

Newer version was released v6.1 using esp board manager 1.0.6. I've included check for the Arduino-esp32 version inside the code that throws error when  Arduino-esp32 version 2.x is detected. I tested it using continues output in the smoothing and 50Hz smoothing rate: for 50Hz works at 0% error rate for around 500 RGBW LEDs setup and at 50% error rate for ~800 RGBW (sk6812 internal data line is limiting this, not fast enough to render it and receive incoming large frames at the same time).

Please test it and let me know if it solved your problem.

Link to 6.1 release: https://github.com/awawa-dev/HyperSerialESP32/releases/tag/v6.1.0.0

**Reply by inferno1231** [reply]
> Please test it and let me know if it solved your problem.
> 
> Link to 6.1 release: https://github.com/awawa-dev/HyperSerialESP32/releases/tag/v6.1.0.0

It works now, thank you very much!
Compiled 4 MB version, works great too.

**Reply by awawa-dev** [reply]
I'm glad it helped. Thank you for your support :)
